content/flux/v0.x/get-started/syntax-basics.md

```diff
@@ -0,0 +1,353 @@
+Flux [basic types](#basic-types).
+
+- [Records](#records)
+- [Arrays](#arrays)
+- [Dictionaries](#dictionaries)
+- [Functions](#functions)
+
+#### Records
+A **record** is a collections of key-value pairs.
+Each key is a string.
+Each value can be a different data type.
+
+```js
+{name:"Jim", age: 42, "favorite color": "red"}
+```
+
+Use **dot notation** or **bracket notation** to access a properties of a record:
+
+{{% note %}}
+Use bracket notation to reference record properties with special or
+white space characters in the property key.
+{{% /note %}}
+
+```js
+o = {name:"Jim", age: 42, "favorite color": "red"}
+
+o.name
+// Returns Jim
+
+o.age
+// Returns 42
+
+o["favorite color"]
+// Returns red
+```
+
+_For more information, see [Work with records](/flux/v0.x/data-types/composite/record/)._
+
+#### Arrays
+An **array** is a collection of values of the same type.
+
+```js
+n = 4
+l = [1,2,3,n]
+
+l
+// Returns [1, 2, 3, 4]
+```
+
+Use **bracket notation** to access a value at a specific index in an array:
+
+```js
+a = ["foo","bar","baz","quz"]
+
+a[0]
+// Returns foo
+```
+
+_For more information, see [Work with arrays](/flux/v0.x/data-types/composite/array/)._
+
+#### Dictionaries
+A **dictionary** is a collection of key-value pairs with keys of the same type
+and values of the same type.
+
+```js
+[1: "foo", 2: "bar"]
+```
+
+Use [`dict.get()`](/flux/v0.x/stdlib/dict/get/) to access elements in a dictionary:
+
+```js
+import "dict"
+
+d = [1: "foo", 2: "bar"]
+
+dict.get(dict: d, key: "1", default: "")
+// Returns foo
+```
+
+_For more information, see [Work with dictionaries](/flux/v0.x/data-types/composite/dict/)._
+
+#### Functions
+A **function** is a block of code that uses a set of parameters to perform an operation.
+Functions can be named or anonymous.
+Define parameters in parentheses (`()`) and pass parameters into an operation
+with the [arrow operator (`=>`)](/flux/v0.x/spec/operators/#function-operators).
+
+```js
+square = (n) => n * n
+
+square(n:3)
+// Returns 9
+```
+
+{{% note %}}
+Flux does not support positional parameters.
+Parameters must always be named when calling a function.
+{{% /note %}}
+
+##### Predicate functions
+Predicate functions use [predicate expressions](#predicate-expressions) to evaluate
+input and return `true` or `false`. For example:
+
+```js
+examplePredicate = (v) => v == "foo"
+
+examplePredicate(v: "foo")
+// Returns true
+
+examplePredicate(v: "bar")
+// Returns false
+```
+
+For more information about working with functions, see:
+
+- [Work with functions](/flux/v0.x/data-types/composite/function/)
+- [Define custom functions](/flux/v0.x/define-functions/)
+
+### Regular expression types
+A **regular expression** is a regular expression pattern used to evaluate strings.
+Use regular expressions in [predicate expressions](#predicate-expressions) or with
+the [`regexp` package](/flux/v0.x/stdlib/regexp/).
+
+```js
+regex = /^foo/
```

Comment:
what does the symbol before foo do?

Reply:
This is a regular expression metacharacter that represents the beginning of a string.